src/main/java/net/sf/jabref/logic/importer/fileformat/MrDLibImporter.java

```diff
@@ -0,0 +1,267 @@
+        String recommendationsAsString = BufferedReaderToParsableString(input);
+        // check for valid format
+        try {
+            SAXParserFactory factory = SAXParserFactory.newInstance();
+            SAXParser saxParser = factory.newSAXParser();
+            DefaultHandler handler = new DefaultHandler() {
+                // No Processing here. Just check for valid xml.
+            };
+
+            try {
+                InputStream stream = new ByteArrayInputStream(recommendationsAsString.toString().getBytes());
+                saxParser.parse(stream, handler);
+            } catch (Exception e) {
+                LOGGER.error(e.getMessage(), e);
+                return false;
+            }
+        } catch (ParserConfigurationException e) {
+            LOGGER.error(e.getMessage(), e);
+            return false;
+        } catch (SAXException e) {
+            LOGGER.error(e.getMessage(), e);
+            return false;
+        }
+        return true;
+    }
+
+    private String BufferedReaderToParsableString(BufferedReader input) throws IOException {
+        String line;
+        StringBuilder sb = new StringBuilder();
+        try {
+            while ((line = input.readLine()) != null) {
+                sb.append(line);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return sb.toString();
+
+    }
+
+    /*
+     * @see net.sf.jabref.logic.importer.Importer#importDatabase(java.io.BufferedReader)
+     */
+    @Override
+    public ParserResult importDatabase(BufferedReader input) throws IOException {
+        // The Bibdatabase that gets returned in the ParserResult.
+        BibDatabase bibDatabase = new BibDatabase();
+        // The document to parse
+        String recommendations = BufferedReaderToParsableString(input);
+        // The list ob BibEntries with its associated rank
+        ArrayList<RankedBibEntry> rankedBibEntries = new ArrayList<>();
+        // The sorted BibEntries gets stored here later
+        List<BibEntry> bibEntries = new ArrayList<>();
+        //Parsing the response with a SAX parser
+        try {
+            SAXParserFactory factory = SAXParserFactory.newInstance();
+            SAXParser saxParser = factory.newSAXParser();
+            DefaultHandler handler = new DefaultHandler() {
+
+                boolean authors;
+                boolean published_in;
+                boolean title;
+                boolean year;
+                boolean snippet;
+                boolean rank;
+                boolean type;
+                String htmlSnippetSingle;
+                int htmlSnippetSingleRank = -1;
+
+                BibEntry currentEntry;
+
+
+                @Override
+                public void startElement(String uri, String localName, String qName, Attributes attributes)
+                        throws SAXException {
+
+                    if (qName.equalsIgnoreCase("related_article")) {
```

Comment:
I am not a huge fan of this huge if else structure. Either you use switch which is efficient on strings, or you could get some inspiration of here:
http://stackoverflow.com/questions/10175805/how-to-avoid-a-lot-of-if-else-conditions